cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -1755,19 +1755,33 @@ public List<Condition> getWarningConditions() {
     }
 
     public DeploymentStrategy getDeploymentStrategy() {
-        if (templateDeploymentStrategy == io.strimzi.api.kafka.model.template.DeploymentStrategy.ROLLING_UPDATE) {
-            return new DeploymentStrategyBuilder()
-                    .withType("RollingUpdate")
-                    .withRollingUpdate(new RollingUpdateDeploymentBuilder()
-                            .withMaxSurge(new IntOrString(1))
-                            .withMaxUnavailable(new IntOrString(0))
-                            .build())
-                    .build();
-        } else {
-            return new DeploymentStrategyBuilder()
-                    .withType("Recreate")
-                    .build();
+        if (templateDeploymentStrategy == null) {
+            return buildRollingUpdateDeploymentStrategy();
+        }
+        switch (templateDeploymentStrategy) {
+            case ROLLING_UPDATE:
+                return buildRollingUpdateDeploymentStrategy();
+            case RECREATE:
+                return buildRecreateDeploymentStrategy();
+            default:
+                return buildRollingUpdateDeploymentStrategy();
         }
+    } 
+
+    private DeploymentStrategy buildRollingUpdateDeploymentStrategy() {
+        return new DeploymentStrategyBuilder()
+                        .withType("RollingUpdate")
+                        .withRollingUpdate(new RollingUpdateDeploymentBuilder()
+                                .withMaxSurge(new IntOrString(1))
+                                .withMaxUnavailable(new IntOrString(0))
+                                .build())
+                        .build();
+    }
+
+    private DeploymentStrategy buildRecreateDeploymentStrategy() {
+        return new DeploymentStrategyBuilder()
+                        .withType("Recreate")
+                        .build();
```

Comment:
Default is RollingUpdate now, because that's how the `templateDeploymentStrategy` property is initialed in this class and it makes sense. These new methods change is needed to avoid repeating the deployment strategy build in all concrete classes.

Reply:
Are you sure about this? I just tried from main branch and it seems to default to RollingUpdate.

Reply:
Also, back to this ... so why is the new code which is much longer and moch more complex better then the old code?

Reply:
You are right, there was a duplication regarding RollingUpdate build between AbstractModel and KafkaExporter. Now I'll use the exact same approach we have in the other concrete classes and revert this method.